### PR TITLE
Ftr: add json generalizer

### PR DESCRIPTION
**Pull request by EnableAsync**
**What this PR does**:
- Ftr: Add JsonGeneralizer
- Tst: Add unit test for JsonrpcGeneralizer
- **Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

This will be used in jsonrpc in the future.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: filter/generic/generalizer/json.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+func (JsonGeneralizer) Realize(obj interface{}, typ reflect.Type) (interface{}, error) {
+	jsonbytes, ok := obj.(string)
+	if !ok {
+		return nil, perrors.Errorf("unexpected type of obj(=%T), wanted is string", obj)
+	}
+
+	// typ represents a struct instead of a pointer
+	for typ.Kind() == reflect.Ptr {
```
Comment (xavier-niu) [inline filter/generic/generalizer/json.go:68]: `typ` could be a pointer, this for loop may cause some problems(I'm not sure). You should test this branch in unit tests.

File: filter/generic/generalizer/json_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	Child               *mockJsonChild
+}
+
+type mockJsonChild struct {
+	Gender, Email, Name string
+	Age                 int
+}
+
+func TestJsonGeneralizer(t *testing.T) {
+	c := &mockJsonChild{
```
Comment (xavier-niu) [inline filter/generic/generalizer/json_test.go:43]: Please add test case for `c` to check if the for loop is necessary.

**Comment by xavier-niu** [comment]
"GSON" key should be added into constants, please refer to `common/constant/key.go` at line 314.

**Comment by xavier-niu** [comment]
Sample for jsonGeneralizer is required as well.

**Review by AlexStocks — APPROVED** [review]

File: filter/generic/generalizer/gson.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+package generalizer
+
+import (
+	"encoding/json"
+	"reflect"
+	"sync"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common/logger"
```
Comment (xavier-niu) [inline filter/generic/generalizer/gson.go:27]: sort import blocks

**Review by xavier-niu — APPROVED** [review]

**Review by cityiron — APPROVED** [review]

**Pull request merged**